social_core/backends/linkedin.py

```diff
@@ -2,11 +2,57 @@
 LinkedIn OAuth1 and OAuth2 backend, docs at:
     https://python-social-auth.readthedocs.io/en/latest/backends/linkedin.html
 """
-from social_core.exceptions import AuthCanceled
+import datetime
+from calendar import timegm
+
+from social_core.backends.open_id_connect import OpenIdConnectAuth
+from social_core.exceptions import AuthCanceled, AuthTokenError
 
 from .oauth import BaseOAuth2
 
 
+class LinkedinOpenIdConnect(OpenIdConnectAuth):
+    """
+    Linkedin OpenID Connect backend. Oauth2 has been deprecated as of August 1, 2023.
+    https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin-v2?context=linkedin/consumer/context
+    """
+
+    name = "linkedin-openidconnect"
+    # Settings from https://www.linkedin.com/oauth/.well-known/openid-configuration
+    OIDC_ENDPOINT = "https://www.linkedin.com/oauth"
+
+    # https://developer.okta.com/docs/reference/api/oidc/#response-example-success-9
+    # Override this value as it is not provided by Linkedin.
+    # else our request falls back to basic auth which is not supported.
+    TOKEN_ENDPOINT_AUTH_METHOD = "client_secret_post"
+
+    def validate_claims(self, id_token):
+        """Copy of the regular validate_claims method without the nonce validation."""
+
+        utc_timestamp = timegm(datetime.datetime.utcnow().utctimetuple())
+
+        if "nbf" in id_token and utc_timestamp < id_token["nbf"]:
+            raise AuthTokenError(self, "Incorrect id_token: nbf")
+
+        # Verify the token was issued in the last 10 minutes
+        iat_leeway = self.setting("ID_TOKEN_MAX_AGE", self.ID_TOKEN_MAX_AGE)
+        if utc_timestamp > id_token["iat"] + iat_leeway:
+            raise AuthTokenError(self, "Incorrect id_token: iat")
+
+        # Skip the nonce validation for linkedin as it does not provide any nonce.
+        # https://stackoverflow.com/questions/76889585/issues-with-sign-in-with-linkedin-using-openid-connect
```

Comment:
I also verified this on my own setup, LinkedIn does indeed not provide a `nonce` in the payload.